packages/grafana-ui/src/components/EmptyState/EmptyState.tsx

```diff
@@ -49,7 +49,7 @@ export const EmptyState = ({
       <div className={styles.container}>
         {!hideImage && imageToShow}
         <Stack direction="column" alignItems="center">
-          <Text variant="h4" textAlignment="center">
+          <Text variant="h4" textAlignment="center" role={variant === 'not-found' ? 'alert' : undefined}>
```

Comment:
I don't think this should be added to the base component. I think this may be an overly broad generalisation which won't be suitable in every place it's used.

I think instead consumers should specify this through a prop.

Reply:
Something like `announceNotFound` (open to suggestions for the name)? I would argue it should default to true at least then? Or do you want to be more generic as in specifying the role?

Reply:
Would this make sense @joshhunt?

Reply:
tbh, I think i would just give it a role / aria-live prop and pass that into the root element. I don't think we get anything by 'abstracting' it here.

Reply:
Okay, but how do you suggest we can adress:
1. The searchbox use cases that we have (kbar and dashboard search)
2. Communicate this to other squads

Would extending the documentation suffice in the second case? Also, how valuable is it to pass it to the root element, especially when there is an alert role? Then you may get superfluous information that is not the message (although some times it may be useful)

I still think this should be the goto solution in this use case, and I would really appreciate a non-hypothetical use case where this doesn't work. If a case like that arises in the future we can always change.

Reply:
> The searchbox use cases that we have (kbar and dashboard search)

We use the prop

> Communicate this to other squads

We document and fix as we go. Not all a11y concerns can be solve automatically - sometimes (often!) consumers will need to do the right thing.

Reply:
I think for the search box use case it makes sense, so maybe that's what we can document.